src/credentials/BUILD.gn

```diff
@@ -37,10 +37,19 @@ static_library("credentials") {
     "examples/DeviceAttestationCredsExample.h",
     "examples/DeviceAttestationVerifierExample.cpp",
     "examples/DeviceAttestationVerifierExample.h",
+    "examples/GroupDataProviderExample.cpp",
   ]
 
   cflags = [ "-Wconversion" ]
 
+  # The default configuration is too small for the group data provider unit tests,
+  # so the defauls must be overwritten wherever the group data provider implementation
+  # is compiled, either in the library, or for each individual application.
+  defines = [
+    "CHIP_CONFIG_MAX_GROUP_ENDPOINTS_PER_FABRIC=8",
+    "CHIP_CONFIG_MAX_GROUPS_PER_FABRIC=6",
+  ]
```

Comment:
This is indeed a headache. The default limits are too strict for the tests, and the overwrites are set by the module that compiles the provider. We could package the provider with the tests, but we'll need to add it to the every application that use it too, which at this point would be every application.

Reply:
I don't understand our build system enough to have good suggestions here.

Ideally, either these constants would not be part of how the provider is compiled or they would be set by the "toplevel" app compilation process and the unit test app would set them...

Reply:
In either case, I think that's beyond the scope of this pull request. Though I had to modify the data provider to accommodate the server cluster, the goal of this PR is to update the groups cluster implementation.

Reply:
It cannot be beyond the scope of a PR to avoid doing something that is liable to cause mysterious breakage.

If the test cannot be run with certain limits, then disable the test when the requirements are not met.